src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/HttpParser.cs

```diff
@@ -9,7 +9,7 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
 {
-    public class HttpParser : IHttpParser
+    public class HttpParser<TRequestHandler> : IHttpParser<TRequestHandler> where TRequestHandler : struct, IHttpHeadersHandler, IHttpRequestLineHandler
```

Comment:
Can we remove the where `TRequestHandler` struct? This is a performance optimization not a correctness thing. It means the test code and benchmarks can avoid those little struct wrappers.